src/Agent.Worker/Build/TfsVCCommandManager.cs

```diff
@@ -236,6 +236,20 @@ private string FormatArguments(FormatFlags formatFlags, params string[] args)
             return string.Join(" ", formattedArgs);
         }
 
+        private string GetEndpointData(ServiceEndpoint endpoint, string name)
+        {
+            var trace = HostContext.GetTrace(nameof(SourceProvider));
```

Comment:
I believe this class already has a Trace property setup and ready to use (inherits from AgentService base class). You should be able to replace all `trace.Info(...)` statements with `Trace.Info(...)`.

If so, you can delete this GetTrace line, otherwise you the nameof(...) needs to be fixed (references an unrelated class).